File: frameworks/webdavclient/src/main/java/org/exoplatform/frameworks/davclient/search/AbstractQuery.java
/***************************************************************************
 * Copyright 2001-2006 The eXo Platform SARL         All rights reserved.  *
 * Please look at license.txt in info directory for more license detail.   *
 **************************************************************************/

package org.exoplatform.frameworks.davclient.search;

import org.exoplatform.frameworks.davclient.request.PropertyList;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

/**
 * Created by The eXo Platform SARL
 * Author : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 * @version $Id: $
 */

public abstract class AbstractQuery implements DavQuery {

  protected PropertyList properties = new PropertyList();  
  
  public void setRequiredProperty(String propertyName) {
    properties.setProperty(propertyName);
  }
  
  public abstract Element toXml(Document xmlDocument);
  
}

File: 1.6/applications/dotnet/exo_jcr.webdav.csclient/Search/AbstractQuery.cs
/***************************************************************************
 * Copyright 2001-2007 The eXo Platform SAS          All rights reserved.  *
 * Please look at license.txt in info directory for more license detail.   *
 **************************************************************************/
using System;
using System.Collections.Generic;
using System.Collections;
using System.Text;
using System.Xml;

/**
 * Created by The eXo Platform SARL
 * Authors : Vitaly Guly <gavrik-vetal@ukr.net/mail.ru>
 *         : Max Shaposhnik <uy7c@yahoo.com>
 * @version $Id:
 */

namespace exo_jcr.webdav.csclient.Search
{
    public abstract class AbstractQuery : DavQuery
    {

        protected ArrayList properties = new ArrayList();

        public void addSelectProperty(String propertyName)
        {
            properties.Add(propertyName);
        }

        public abstract void toXml(XmlTextWriter writer);

    }
}
